docs/sql-ref-number-pattern.md

```diff
@@ -19,4 +19,137 @@ license: |
   limitations under the License.
 ---
 
-TODO: Add the content of Number Patterns for Formatting and Parsing
+### Description
+
+Functions such as `to_number` and `try_to_number` support converting between values of string and
+Decimal type. Such functions accept format strings indicating how to map between these types.
+
+### Syntax
+
+Number format strings support the following syntax:
+
+```
+  { ' [ S ] [ L | $ ] 
```

Comment:
do we end up supporting `L`?

Reply:
Good catch, no, we did not. Removed this part from the syntax.